Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ps1

```diff
@@ -1,134 +1,142 @@
-param
-(
-    [String] [Parameter(Mandatory = $true)]
-    $ConnectedServiceName,
-
-    [String] [Parameter(Mandatory = $true)]
-    $WebAppName,
+[CmdletBinding()]
+param()
+
+Trace-VstsEnteringInvocation $MyInvocation
+
+try{
+
+	# Get inputs.
+	$WebAppName = Get-VstsInput -Name WebAppName -Require
+	$DeployToSlotFlag = Get-VstsInput -Name DeployToSlotFlag -Require
+	$ResourceGroupName = Get-VstsInput -Name ResourceGroupName
+	$SlotName = Get-VstsInput -Name SlotName
+	$Package = Get-VstsInput -Name Package -Require
+	$RemoveAdditionalFilesFlag = Get-VstsInput -Name ScriptArgument
+	$ExcludeFilesFromAppDataFlag = Get-VstsInput -Name ExcludeFilesFromAppDataFlag 
+	$TakeAppOfflineFlag = Get-VstsInput -Name TakeAppOfflineFlag 
+	$VirtualApplication = Get-VstsInput -Name VirtualApplication 
+	$AdditionalArguments = Get-VstsInput -Name AdditionalArguments 
+	$WebAppUri = Get-VstsInput -Name WebAppUri
+	$SetParametersFile = Get-VstsInput -Name SetParametersFile
+
+	# Initialize Azure.
+
+	Import-Module $PSScriptRoot\ps_modules\VstsAzureHelpers_
+	Initialize-Azure
+
+	# Import the loc strings.
+	Import-VstsLocStrings -LiteralPath $PSScriptRoot/Task.json
+
+
+	Write-Verbose "Starting AzureRM WebApp Deployment Task"
```

Comment:
Trace-VstsEnteringInvocation already indicates this. One thing we focused on with the new task lib is built-in tracing. Use the lib, get tracing for free.